### DISCUSSION TITLE
Full amount of data on each drive in distributed mode.

### DISCUSSION THREAD

**Discussion by jreoka** [minio/minio#13427] [Q&A] [answered]
## Expected Behavior

The data should at most be a bit over twice the actual amount of imported data.

## Current Behavior
In my case, every drive contains over the original full amount of data in the bucket.

## Possible Solution
No idea.

## Steps to Reproduce (for bugs)

1.
2.
3.
4.

## Context
I am testing Minio by selecting four destinations on the same drive, but I doubt this would affect its behavior like this.

## Regression

No

## Your Environment

* Version used (`minio --version`): Latest Minor Release as of the date of this post.
* Server setup and configuration: Using rclone s3 mode to import 1.3 TB of data to Minio.
* Operating System and version (`uname -a`): Linux cross-server 5.14.9-051409-generic #202109300934 SMP Thu Sep 30 09:39:33 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

https://i.imgur.com/D4cDYCV.png <--- Image of the issue.

**Comment by harshavardhana** [comment]
@DillFrescott that doesn't make any sense. Please describe your setup in detail and how you found out they are same.

**Comment by jreoka** [comment]
@harshavardhana It's four "drives" for minio but they are actually on the same drive. The bucket has one TB of data in it. It is the only bucket. The drive I am hosting the data on only has the minio files on it. I have m1, m2, m3, and m4 as the folders minio is using to get it to turn on erasure coding mode.

There is only 1 TB of data in the bucket but m1 is a TB large. m2 is a TB large. m3 is a TB large, and m4 is a TB large.

It somehow duplicated the full amount of data for each folder...

I am failing to understand how that doesn't make sense.

**Comment by jreoka** [comment]
I figured out they are the same by running `du -hs` command on each of them. They are all around 900 - 1000 GB

**Comment by harshavardhana** [comment]
> I figured out they are the same by running `du -hs` command on each of them. They are all around 900 - 1000 GB

how did you start MinIO @DillFrescott ?

**Comment by jreoka** [comment]
MINIO_ROOT_USER=cross MINIO_ROOT_PASSWORD=mypassword ./minio server --console-address ":7000" /media/hdd/miniodata/m1 /media/hdd/miniodata/m2 /media/hdd/miniodata/m3 /media/hdd/miniodata/m4

Thats the exact command minus my actual password. :)

**Comment by harshavardhana** [comment, marked as answer]
Here is how I tested, 

```
~ minio server /tmp/xl{1...4}

Verifying if 1 bucket is consistent across drives...
Automatically configured API requests per node based on available memory on the system: 82
Status:         4 Online, 0 Offline. 
API: http://10.0.0.67:9000  http://172.16.3.2:9000  http://172.17.0.1:9000  http://172.18.0.1:9000  http://127.0.0.1:9000                   
RootUser: minio 
RootPass: minio123 
```

```
~ mc mb myminio/testbucket
~ dd if=/dev/zero of=bigfile bs=1M count=256
256+0 records in
256+0 records out
268435456 bytes (268 MB, 256 MiB) copied, 0.0889448 s, 3.0 GB/s
~ mc cp bigfile myminio/testbucket/prefix/
bigfile:                    256.00 MiB / 256.00 MiB ┃▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓┃ 255.30 MiB/s 
```

```
~ mc ls myminio/testbucket/prefix/
[2021-10-12 21:26:30 PDT] 256MiB bigfile
```

```
~ du -sh /tmp/xl*/testbucket/prefix
129M    /tmp/xl1/testbucket/prefix
129M    /tmp/xl2/testbucket/prefix
129M    /tmp/xl3/testbucket/prefix
129M    /tmp/xl4/testbucket/prefix
```

> 256*2 = 512/4

This is how it's used, I don't see any problems here. 

We have zillions of deployments out there - you should figure out locally what you did. This looks like a simple setup issue. 

Moving this to discussion if you wish to discuss further, but its not something we think is an issue.

**Comment by jreoka** [comment]
@harshavardhana I figured it out. You have to mount the drives using /tmp/xl{1...4} as an example. You can't specify each path separately apparently.

**Reply by jreoka** [reply]
Thats what I was doing wrong
